boards/arm/stm32l562e_dk/stm32l562e_dk_ns.dts

```diff
@@ -15,8 +15,8 @@
 	#size-cells = <1>;
 
 	chosen {
-		zephyr,console = &usart1;
-		zephyr,shell-uart = &usart1;
+		zephyr,console = &lpuart1;
+		zephyr,shell-uart = &lpuart1;
```

Comment:
This PR works but is not ideal. After TZ is enabled, the log output over usart1 is garbage from the non secure Zephyr application. I have used lpuart1 as a workaround but it is a bit awkward because I have to jump wire to wire out the lpuart1 TX/RX pins on stm32l562e_dk board, and the logs for mcuboot and secure firmware come out of usart1 while the log for non secure application comes out of lpuart1.

I have tried to avoid the GPIOA pin control in non secure target since GPIOA is a Trustzone protected peripheral after TZEN is enabled.

I ran the TFM sample application from the STM32Cube_FW_L5 package, the logging seems to work in non secure image, but I don't see anything special in the UART(COM) port initialization code.

After I switch to lpuart1, the logging in non secure application is working, but I don't understand the different since both lpuart1 and usart1 are securable peripherals, and are non-secure after reset.

I have spent quite some time to debug this issue but no avail so far. Maybe can @erwango @ABOSTM shed some light on this?

Reply:
@yestin from what I see this is specific to this dk board and this works on the nucleo. I guess this is due to console confugration for this board on TFM side. Maybe @jamike you can confirm this ?

Reply:
@yestin just to be sure: have you tried to keep uart1 as NS console **without pinmuxing enabled**

Reply:
I believe this commit ([84d97dd8](https://github.com/zephyrproject-rtos/zephyr/pull/31118/commits/84d97dd894571bb0993c437e8effd71c69242f38)) shall disable the pinmuxing for the uart1 as NS console. After build, I checked the generated dts file build/zephyr/zephyr.dts and did not see the pinmux configuration.

Reply:
> I believe this commit ([84d97dd8](https://github.com/zephyrproject-rtos/zephyr/pull/31118/commits/84d97dd894571bb0993c437e8effd71c69242f38)) shall disable the pinmuxing for the uart1 as NS console. After build, I checked the generated dts file build/zephyr/zephyr.dts and did not see the pinmux configuration.

And does it allow to get working console on uart1 ?

Reply:
ok, I had a test and I confirm I'm seeing the same thing.

Reply:
Though, at first sight, I'd rather suspect a clock issue.

Reply:
So I manage to make ns console work on uart1 by setting `current-speed = <57600>;`.
This confirms it is a clock issue:
TFM and Zephyr doesn't configure the same clock tree for usart1.
Since we don't face this issue on lpuart1, I suspect the misalignment either comes from APB2 PRESC or USART1SEL(RCC_BDCR).

Reply:
This is good stuff. I was able to see the ns console working with your suggestion.

Unfortunately, I think I lost my stm32l562e_dk board when I was trying to deactivate TZEN. I was able to recover it last time by applying 3V3 on PH3-BOOT0 signal to make it boot from RSS, but not this time. 😞 

I guess setting RDP level to 1 before TZEN deactivation was not a good idea. Anyway, I will not be able to debug any further on this for some time.

Reply:
I faced some issue with the board myself during the investigation.
I managed to get it back using stm32cubeprogrammer, by erasing the flash fully and then programming a hello world (using stm32cubeprogrammer UI).

Reply:
Should be fixed by https://github.com/zephyrproject-rtos/zephyr/pull/31178

Reply:
Thanks for the fix. I have rebased on top of your fix and verified the new patch that the logging works well on usart1 now.